File: 3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.cpp
#include <ranges>

static auto init = []() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);
  return nullptr;
}(); 

class Solution {
public:
  int minOperations(vector<int>& nums, int k) {
    auto low_values = nums
                      | views::filter([k](auto num) { return num < k; })
                      | views::transform([](auto num) {
                            return static_cast<long long>(num); }) ;
    auto q = priority_queue(ranges::begin(low_values), ranges::end(low_values),
                            greater{});
    auto num_ops = 0;
    while (!empty(q)) {
      ++num_ops;
      if (size(q) == 1) break;
      auto x = q.top(); q.pop();
      auto y = q.top(); q.pop();
      auto new_val = (min(x, y) << 1 ) + max(x, y);
      if (new_val < k) q.push(new_val);
    }
    return num_ops;
  }
};

File: 3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.java
class Solution {
    public int minOperations(int[] nums, int k) {
        PriorityQueue<Long> queue = new PriorityQueue<>();
        int operations = 0;
        for(int a = 0; a < nums.length; a++) {
            queue.add((long)nums[a]);
        }
        while(queue.size() > 1 && queue.peek() < k) {
            long val1 = queue.remove();
            long val2 = queue.remove();
            queue.add((val1*2) + val2);
            operations = operations + 1;
        }
        return operations;
    }
}